java/org.travlyn.server/src/main/java/org/travlyn/server/apiAccess/APIRequest.java

```diff
@@ -0,0 +1,47 @@
+package org.travlyn.server.apiAccess;
+
+import org.travlyn.server.util.Pair;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Set;
+
+/**
+ * Generic utility class to fetch data from API via GET request.
+ *
+ * @author Joshua Schulz
+ * @since 1.0
+ */
+public class APIRequest {
+    private final URL requestURL;
+
+    public APIRequest(String apiURL, Set<Pair<String, String>> parameters) throws MalformedURLException {
+        String completeURL = apiURL + ParameterStringBuilder.getParamString(parameters);
+        this.requestURL = new URL(completeURL);
+    }
```

Comment:
Am I correct that this class can only handle GET Requests? Not sure if this is applicable, but wouldn't it be useful to reuse this class on client-side, as the client also has to access our API? It might get to complex.

If you change the second parameter to `Set<Pair<String, Serializable>> parameters`, you also can pass integers as parameters and you do not need to cast them. I guess this would also fix the problem, that you have to use a `Pair<...>`

Reply:
Problem with using a Map is not the cast or something, but that i can not specify to equivalent keys and keys are used as parameter names for API request. This leads to the fact that i can not use the same API parameter twice, but this is need for DBpedia request.